futures-util/src/stream/stream/mod.rs

```diff
@@ -544,6 +548,40 @@ pub trait StreamExt: Stream {
         Flatten::new(self)
     }
 
+    /// Maps a stream like [`StreamExt::map`] but flattens nested `Stream`s.
+    ///
+    /// [`StreamExt::map`] is very useful, but if it produces a `Stream` instead,
```

Comment:
I'd rephrase this to avoid talking about "extra layers"`-- perhaps refer to `.map(...).flatten()`? Also, consider mentioning that this combinator will yield all items from the first stream before coming back and pulling out another stream, so it's not suitable for handling values from multiple streams concurrently.

Reply:
@cramertj do we need some iterator which acts like `flat_map` but pulls streams concurrently?

Reply:
Absolutely-- I want it all the time, and opened https://github.com/rust-lang/futures-rs/issues/1781 for this. Unfortunately it's somewhat complicated to implement efficiently, and would require something similar to `FuturesUnordered` for streams.